pmlc/conversion/stdx_to_llvm/stdx_to_llvm.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020, Intel Corporation
+
+#include "pmlc/conversion/stdx_to_llvm/stdx_to_llvm.h"
+
+#include "mlir/ADT/TypeSwitch.h"
+#include "mlir/Conversion/LoopToStandard/ConvertLoopToStandard.h"
+#include "mlir/Conversion/StandardToLLVM/ConvertStandardToLLVM.h"
+#include "mlir/Conversion/StandardToLLVM/ConvertStandardToLLVMPass.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/StandardOps/Ops.h"
+#include "mlir/Pass/Pass.h"
+
+#include "pmlc/dialect/stdx/ir/ops.h"
+
+using namespace mlir;  // NOLINT[build/namespaces]
+
+namespace pmlc::conversion::stdx_to_llvm {
+
+namespace stdx = dialect::stdx;
+
+namespace {
+
+template <typename SourceOp>
+class LLVMLegalizationPattern : public LLVMOpLowering {
+ public:
+  explicit LLVMLegalizationPattern(LLVMTypeConverter& converter)
+      : LLVMOpLowering(SourceOp::getOperationName(), converter.getDialect()->getContext(), converter) {}
+
+ protected:
+  LLVM::LLVMDialect& getDialect() const { return lowering.getDialect(); }
+};
+
+Optional<LLVM::AtomicBinOp> matchBinOp(stdx::AtomicRMWOp op) {
+  auto* body = op.getBody();
+  auto* terminator = body->getTerminator();
+  auto yieldOp = llvm::cast<stdx::AtomicRMWYieldOp>(terminator);
+  auto defOp = yieldOp.result().getDefiningOp();
```

Comment:
We need to additionally should insure that the 'Add' or whatever has one of it's inputs match the argument to the region and that it is the sole instruction in the loop besides the yield (not part of something more complex, which needs the CmpXchg lowering).

Reply:
Yep, still WIP, will address after I get this RFC published